### PR TITLE
Filename fixes

### LINKED ISSUE #1144: Strange behaviour with field "file"
**Elasticsearch version**: 6.8.2

**Moloch version**: 2.0.0

**OS name and version**: Ubuntu 16.04

**How was Moloch built/installed**: deb

**Provide logs, stack traces and steps to reproduce**:

* The query "file == test.pcap" display events from this pcap files but the query "file != test.pcap" or "file == EXISTS!" always returns "no results" even if there are other offline pcap on ES.

* "Filename" column is always empty if i display it

* "Export Unique Files" and "Export Unique Files with counts" commands both return a blank page

I encounter same problems on version 1.8.0

### PR DESCRIPTION
**Pull request by 31453**
fixes #1144

### CODE REVIEW

**Comment by 31453** [comment]
How can I make these tests better? I'm not happy with the ones in api-sessions.t or api-spigraph.t because they will break if we add more pcaps. HALP @awick!

File: tests/api-sessions.t [hunk 1]
```diff
@@ -146,3 +146,11 @@ tcp,1386004309468,1386004309478,10.180.156.185,53533,US,10.180.156.249,1080,US,2
 # no map data
     $json = viewerGet("/sessions.json?startTime=1386004308&stopTime=1386004400&facets=1&expression=" . uri_escape("file=$pwd/bigendian.pcap|file=$pwd/socks-http-example.pcap|file=$pwd/bt-tcp.pcap"));
     eq_or_diff($json->{map}, from_json('{}'), "no map data");
+
+# Check file != blah.pcap
+    my $json = viewerGet("/sessions.json?date=-1&expression=" . uri_escape("file!=$pwd/https-connect.pcap"));
```
Comment (awick) [inline tests/api-sessions.t]: chose several files here and do 2 tests
1) file==file1 && file==file2 && file==file3
2) file!=file1 && file==file2 && file==file3

File: tests/api-sessions.t [hunk 2]
```diff
@@ -146,3 +146,11 @@ tcp,1386004309468,1386004309478,10.180.156.185,53533,US,10.180.156.249,1080,US,2
...
+    is ($json->{recordsFiltered}, 271, "file !=");
+
+# Check file == EXISTS!
+    my $json = viewerGet("/sessions.json?date=-1&expression=file=EXISTS!");
```
Comment (awick) [inline tests/api-sessions.t]: same, do
file=EXISTS!&&file=file1&&file=file2&&file=file3

File: tests/api-spigraph.t [hunk 3]
```diff
@@ -141,3 +141,7 @@ my ($json, $mjson);
 # no map data
     $json = viewerGet("/spigraph.json?date=-1&field=http.useragent&expression=" . uri_escape("file=$pwd/socks5-reverse.pcap|file=$pwd/socks-http-example.pcap|file=$pwd/bt-tcp.pcap"));
     eq_or_diff($json->{map}, from_json('{}'), "no map data");
+
+# file field works
+    $json = viewerGet("/spigraph.json?date=-1&field=fileand");
+    cmp_ok ($json->{recordsFiltered}, '==', 272);
```
Comment (awick) [inline tests/api-spigraph.t]: you could do >= instead of == here or add a expression file1&file2&file3 again

File: tests/api-unique.t [hunk 4]
```diff
@@ -207,3 +207,17 @@ fe80::211:25ff:fe82:95b5.0, 2
...
+eq_or_diff($txt,
+"/Users/erinne23/Documents/dev/moloch/tests/pcap/socks-http-example.pcap, 3
+/Users/erinne23/Documents/dev/moloch/tests/pcap/socks-http-pass.pcap, 3
+/Users/erinne23/Documents/dev/moloch/tests/pcap/socks-https-example.pcap, 3
+/Users/erinne23/Documents/dev/moloch/tests/pcap/socks5-http-302.pcap, 1
+/Users/erinne23/Documents/dev/moloch/tests/pcap/socks5-rdp.pcap, 1
+/Users/erinne23/Documents/dev/moloch/tests/pcap/socks5-reverse.pcap, 1
+/Users/erinne23/Documents/dev/moloch/tests/pcap/socks5-smtp-503.pcap, 1
+/Users/erinne23/Documents/dev/moloch/tests/pcap/v6-http.pcap, 6
+", "filename counts");
```
Comment (awick) [inline tests/api-unique.t]: you would use $pwd instead of /Users..*pcap here.

**Pull request merged**